File: April-2023/Java/KidsWithTheGreatestNumberOfCandies.java
// 1431. Kids With the Greatest Number of Candies
// get max and compare with max
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> l = new LinkedList<>();
        int max=0;
        for(int n: candies){            
                max=Math.max(max,n);                            
            }
        int i;
        for(i=0;i<candies.length;i++){
       if(candies[i]+extraCandies>=max){
                l.add(true);                              
                }
            else{
                l.add(false);
                }            
            }
        return l;
    }
}

File: April-2023/C++/KidsWithTheGreatestNumberOfCandies.cxx
// 1431. Kids With the Greatest Number of Candies
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> l;
        int max = 0;
        for (int n : candies) {
            max = std::max(max, n);
        }
        for (int i = 0; i < candies.size(); i++) {
            if (candies[i] + extraCandies >= max) {
                l.push_back(true);
            }
            else {
                l.push_back(false);
            }
        }
        return l;
    }
};
